internals/src/array_vec.rs

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! A simplified `Copy` version of `arrayvec::ArrayVec`.
+
+use core::fmt;
+
+pub use safety_boundary::ArrayVec;
+
+/// Limits the scope of `unsafe` auditing.
+mod safety_boundary {
+    use core::mem::MaybeUninit;
+    use crate::const_tools::cond_const;
+
+    /// A growable contiguous collection backed by array.
+    #[derive(Copy)]
+    pub struct ArrayVec<T: Copy, const CAP: usize> {
+        len: usize,
+        data: [MaybeUninit<T>; CAP],
+    }
+
+    impl<T: Copy, const CAP: usize> ArrayVec<T, CAP> {
```

Comment:
In 6c832cf55b83ae6cf1765c37860490e855904551:

We have `T: Copy` on every single method in this module. Do we want to weaken that? I imagine that `Copy` is not always needed, e.g. for `as_slice`, `as_mut_slice`, `Deref`, etc.

Reply:
It still cannot be constructed without causing memory leaks (and possibly soundness issues if more complicated code is added in the future), so the only thing it'd help with is us writing a bit less characters.

Reply:
Ah, I see. All the methods *work* without `Copy` but internally our values are inside `MaybeUninit`, which won't drop the contained value. (It looks like we can manually do it using `assume_init` or the like, but (a) these aren't in our MSRV; (b) we'd have to implement them in a `Drop` implementation, which would mean we couldn't be `Copy`.....presumably why the real `ArrayVec` isn't `Copy`.

Reply:
> presumably why the real `ArrayVec` isn't `Copy`.

Exactly, everything else is doable.